find-trio-denovo.py

```diff
@@ -14,167 +13,105 @@
-				elif firstCol == "child" and secondCol == "dad":
-					(chrom, pos, ID, ref, alt, qual, Filter, info, format, childgeno, dadgeno, momgeno)= line.strip("\n").split("\t")
-				elif firstCol ==  "child" and secondCol == "mom":
-					(chrom, pos, ID, ref, alt, qual, Filter, info, format, childgeno, momgeno, dadgeno)= line.strip("\n").split("\t")
-
-				# split each of the dad/mom/child columns by ":", to access only the genotype
-				# first element of the list is the genotype when format is Genotype:Quality:ReadDepth:etc.
-				dadgeno_list = dadgeno.split(":")
-				dadGeno = (dadgeno_list[0])
-				
-				momgeno_list = momgeno.split(":")
-				momGeno = (momgeno_list[0])
-				
-				childgeno_list = childgeno.split(":")
-				childGeno = (childgeno_list[0])
-				
-				#now split the genotypes into individual alleles
-				#will split on "/" or "|"
-				(dadAllele1, dadAllele2) = re.split(r"/|\|",dadGeno)	    #to apply to both phased 
-				(momAllele1, momAllele2) = re.split(r"/|\|",momGeno)	    #and unphased entries
-				(childAllele1, childAllele2) = re.split(r"/|\|",childGeno)
-
-				#checks which delimiter was used.  | = phased .   / = unphased
-				#assumes that delimiter on dadGeno is same as momGeno and childGeno 
-				dialect = csv.Sniffer().sniff(dadGeno,["/","|"])
-				delim = (dialect.delimiter)
-
-				#create two lists; one containing both child alleles, the other containing all parent alleles
-				childAlleles = [childAllele1, childAllele2]
-				parentAlleles = [dadAllele1, dadAllele2, momAllele1, momAllele2]
-				
-				child1_in_dad = 0   #to track whether alleles match or not
-				child2_in_dad = 0   
-				child1_in_mom = 0
-				child2_in_mom = 0
-
-				#want to check for each child allele in all parent alleles
-				for i in range(0,2):   
-					for j in range(0,4):  #will check all parent alleles for a match; indices 0-3
-						if childAlleles[i] == parentAlleles[j]:
-						#parentAlleles[0] and parentAlleles[1] are dad's 1st and 2nd alleles
-						#parentAlleles[2] and parentAlleles[3] are mom's 1st and 2nd alleles
-							
-							#check child's first allele
-							if i==0 and j==0: 
-								child1_in_dad = 1
-							elif i==0 and j==1:
-								child1_in_dad = 1
-							elif i==0 and j==2:
-								child1_in_mom = 1
-							elif i==0 and j==3:
-								child1_in_mom = 1
-	 						
-	 						#check child's second allele
-							elif i==1 and j==0: 
-								child2_in_dad = 1
-							elif i==1 and j==1:
-								child2_in_dad = 1
-							elif i==1 and j==2:
-								child2_in_mom = 1
-							elif i==1 and j==3:
-								child2_in_mom = 1
-
-				#The following conditions were written when I was taking all 4 nucleotides into account
-				#In a VCF, it will usually only be comparing a "0" (ref) or "1" (alt)
-				#It still works, but might be more than what's necessary
-
-				# Variant conditions if the data is phased
-				if delim == "|":
```

Comment:
instead of working with a raw value throughout your code, work with abstractions. Below I turned this into a `phased` variable that is a boolean (`True` / `False` -- Phased / Unphased).